.github/workflows/check_interface_compatibility.yml

```diff
@@ -0,0 +1,83 @@
+name: Check interface compatibility
```

Comment:
Actually I think we need a way to run this easily locally, e.g. `make compat`. Currently it'll probably get annoying to not be able to reproduce and have big comments in PRs.

We could store the JSON in the repository and compare against that, no?